esp-hal/src/dma/mod.rs

```diff
@@ -1692,9 +1686,60 @@ where
         }
     }
 
+    /// Converts a blocking channel to an async channel.
+    pub(crate) fn into_async(mut self) -> ChannelRx<'a, Async, CH> {
+        if let Some(handler) = self.rx_impl.async_handler() {
+            self.set_interrupt_handler(handler);
+        }
+        ChannelRx {
+            burst_mode: false,
```

Comment:
I'm guessing there's a reason you didn't do this?
```suggestion
            burst_mode: self.burst_mode,
```

This is the only thing that's functionally different.